lib/parser.rb

```diff
@@ -16,6 +16,20 @@ def print_first_line
   def print_info
     file = File.foreach(@file)
     file_lines = file.count
-    JSON.pretty_generate({@file => {lines: file_lines}}) 
+
+    new_file = File.open(@file)
+    array = new_file.readlines
+    client_lines = array.grep(/.......ClientUserinfoChanged/)
+    new_file.close
+
+    players_arr = Array.new
+
+    client_lines.map do |line|
+      splitted_lines = line.split('n\\')
+      players_lines = splitted_lines[1]
+      player = players_lines.split('\\').first
+      players_arr << player
+    end
```

Comment:
```suggestion
    players_arr = []

    data = File.open(@file, &:readlines)
    client_lines = data.grep(/.......ClientUserinfoChanged/)

    client_lines.each do |line|
      players_lines = line.split('n\\')[1]
      players_arr << players_lines.split('\\').first
    end
```
Eu dei uma leve refatorada no teu código (não testei, mas deve funcionar). Alguns pontos:

1. Seria interessante você dar uma lida a mais sobre `each` e `map`. Nãoa havia necessidade de usar map nesse caso, já que você queria somente iterar sobre as linhas (é pra isso que serve o each).
2. Declaração de variável é recomendado que seja sempre no início de um bloco (facilita a legibilidade)
3. Você criou algumas variáveis que eu achei um pouco desnecessárias dentro do bloco, então por isso de uma reduzida no código removendo algumas delas. Lembre-se que cada variável é um espaço na memória que você vai tá alocando, seja prudente com o uso delas kkkk
4. Você pode pegar o conteúdo de um arquivo e armazenar em uma variável e utilizar essa variável pra manipular o conteúdo. Vi que você abria o arquivo, pegava as linhas, ai alaterava um pouco e só depois fechava o arquivo. Não tem necessidade disso também. Fica a dica ai